Source/Core/Core/AchievementManager.cpp

```diff
@@ -246,6 +252,37 @@ void AchievementManager::DoFrame()
   }
 }
 
+bool AchievementManager::CanPause()
+{
+  u32 frames_to_next_pause = 0;
+  bool can_pause = rc_client_can_pause(m_client, &frames_to_next_pause);
+  if (!can_pause)
+  {
+    OSD::AddMessage("Cannot spam pausing in hardcore mode.", OSD::Duration::VERY_LONG,
+                    OSD::Color::RED);
+    OSD::AddMessage(
+        fmt::format("Can pause in {} seconds.", static_cast<float>(frames_to_next_pause) / 60),
+        OSD::Duration::VERY_LONG, OSD::Color::RED);
+  }
+  return can_pause;
+}
+
+void AchievementManager::DoIdle()
+{
+  std::thread([]() {
+    while (Core::GetState(*AchievementManager::GetInstance().m_system) == Core::State::Paused &&
```

Comment:
I played around with this a little bit and crashed. Turns out `m_system` can be nullptr here! Which makes sense if you look at `CloseGame()`. May I suggest changing this loop to:

```
  std::thread([this]() {
    while (true)
    {
      Common::SleepCurrentThread(1000);
      std::lock_guard lg{m_lock};
      if (!m_system || Core::GetState(*m_system) != Core::State::Paused)
        return;
      if (!m_background_execution_allowed)
        return;
      if (!m_client || !IsGameLoaded())
        return;
      rc_client_idle(m_client);
    }
  }).detach();
```